addons/mail/static/src/models/file_uploader.js

```diff
@@ -118,6 +126,8 @@ registerModel({
                 }));
             }
             const attachments = [];
+            const attachmentsTemp = [];
+            const fromDraft = chatter && chatter.webRecord && chatter.webRecord.data && chatter.webRecord.data.state === 'draft';
```

Comment:
`thread.isTransient` is equivalent to draft mode. This is shorter and probably covers more than `data.state === 'draft'`

Reply:
It doesn't seem to be available on `thread` in 16.0

Reply:
In 16.0 `thread.id <= 0` is equivalent

Reply:
I cannot reproduce any case where `thread.id <= 0`.
Even on a newly created record, its value is greater than 0.
Am I missing something?

Reply:
This is when the thread is not actually created, so when you click "New" without saving it, then the `thread.id <= 0`. The actual saving turns into a genuine thread that has a positive id.

By limiting the fix to draft threads / `Creating a new record...` threads, this ensures the regression on uploading state not being properly updated only affects the case when uploading files from a draft record.

The bug is specifically on uploading files on a draft thread, so it makes sense to target the fix on them specifically. This is especially true when the fix comes with a regression that we should avoid adding it in non-buggy scenarios, i.e. uploading files on a non-draft record.

Reply:
I don't know if it is due to the autosave feature or something else, but when I click on "New" (for a vendor bill) and directly drag and drop files into the chatter, `thread` has a positive id. :thinking: 

> The bug is specifically on uploading files on a draft thread

In fact, the bug happens on any record where there is a preview, regardless the state of the thread or the record.